actionview/lib/action_view/template.rb

```diff
@@ -270,13 +270,19 @@ def compile!(view)
             compile(mod)
           end
 
-          # Just discard the source if we have a virtual path. This
-          # means we can get the template back.
-          @source = nil if @virtual_path
           @compiled = true
         end
       end
 
+      class LegacyTemplate < DelegateClass(Template) # :nodoc:
```

Comment:
What is the plan to this class? We will always have a LegacyTemplate or we plan to remove it some day?

Reply:
I think we can remove it eventually.  I want to change [this call](https://github.com/rails/rails/blob/2169bd3d2a9d2f331a5dd6e41d9d638e0da6117c/actionview/lib/action_view/template.rb#L307) to pass in the view and the source.  But for backwards compatibility we need to pass an object that has a reference to the mutated source.

I'll follow up with another commit that deprecates arity one blocks